### PR TITLE
[bsp/allwinner] feat: porting to RT_USING_DEVICE_OPS

### PR DESCRIPTION
**Pull request by polarvid**
## 拉取/合并请求描述：(PR description)

[

#### 为什么提交这份PR (why to submit this PR)

This patch ports the codebase to use the RT_USING_DEVICE_OPS structure, which is required by v5.1.0 Smart kernel, improves modularity and makes it easier to manage device operations by consolidating them into a single structure, enhancing maintainability and future scalability.

#### 你的解决方案是什么 (what is your solution)

Changes:
- Added RT_USING_DEVICE_OPS conditionals to partition.c and drv_sdmmc.c.
- Defined rt_device_ops structures for partition and sdmmc drivers.
- Updated device initialization to use the ops structure if defined.
- Replaced direct function calls with rt_dev_control, rt_dev_read, and rt_dev_write macros where applicable.
- Removed redundant us_delay function from os.c.

#### 请提供验证的bsp和config (provide the config and bsp) 

- BSP: allwinner d1s

- .config:

- action:

]

### 当前拉取/合并请求的状态 Intent for your PR

必须选择一项 Choose one (Mandatory):

- [ ] 本拉取/合并请求是一个草稿版本 This PR is for a code-review and is intended to get feedback
- [x] 本拉取/合并请求是一个成熟版本 This PR is mature, and ready to be integrated into the repo

### 代码质量 Code Quality：

我在这个拉取/合并请求中已经考虑了 As part of this pull request, I've considered the following:

- [x] 已经仔细查看过代码改动的对比 Already check the difference between PR and old code
- [x] 代码风格正确，包括缩进空格，命名及其他风格 Style guide is adhered to, including spacing, naming and other styles
- [x] 没有垃圾代码，代码尽量精简，不包含`#if 0`代码，不包含已经被注释了的代码 All redundant code is removed and cleaned up
- [x] 所有变更均有原因及合理的，并且不会影响到其他软件组件代码或BSP All modifications are justified and not affect other components or BSP
- [x] 对难懂代码均提供对应的注释 I've commented appropriately where code is tricky
- [x] 代码是高质量的 Code in this PR is of high quality
- [x] 已经使用[formatting](https://github.com/mysterywolf/formatting) 等源码格式化工具确保格式符合[RT-Thread代码规范](https://github.com/RT-Thread/rt-thread/blob/master/documentation/contribution_guide/coding_style_cn.md) This PR complies with [RT-Thread code specification](https://github.com/RT-Thread/rt-thread/blob/master/documentation/contribution_guide/coding_style_en.md)

### CODE REVIEW

**Comment by polarvid** [comment]
@ylyamin you could check the PR message or the commit log for details 😉

File: bsp/allwinner/libraries/drivers/sdmmc/drv_sdmmc.h [hunk 1]
```diff
@@ -27,4 +27,18 @@ struct dev_sdmmc
 void sd_mmc1_init(void);
 void sd_mmc1_deinit(void);
 
+#ifdef RT_USING_DEVICE_OPS
+#define rt_dev_has_control(dev)         (dev)->ops->control
```
Comment (BernardXiong) [inline bsp/allwinner/libraries/drivers/sdmmc/drv_sdmmc.h:31]: 这些使用rt_device_* API是否更合适些？

Comment (polarvid) [reply bsp/allwinner/libraries/drivers/sdmmc/drv_sdmmc.h:31]: rt_device API 有特殊的操作协议 （open, init），但是这套 API 没有按照这种方式。

因此建议是建个 issue，后续工作中继续推进驱动的 API 对接问题。本 PR 还是保留宏的形式。

**Comment by polarvid** [comment]
tested with commit https://github.com/RT-Thread/rt-thread/commit/c7e1982215acb516e4c86b17999b512f23e2557f on 柿饼 M7

image

**Review by BernardXiong — APPROVED** [review]

**Pull request merged**